### PR TITLE
Gosec fixies

### PR DESCRIPTION
**Pull request by nplanel**

### CODE REVIEW

File: flow/probes/ovsmirror.go [hunk 1]
```diff
@@ -447,6 +449,9 @@ func (o *ovsMirrorProbe) OnError(err error) {
 			}
 			return false
 		})
+		if err != nil {
```
Comment (dubek) [inline flow/probes/ovsmirror.go]: s/err/errUpdate/ ?

Comment (nplanel) [reply flow/probes/ovsmirror.go]: yep thanks

File: flow/probes/pcapsocket.go [hunk 2]
```diff
@@ -132,11 +132,16 @@ func (p *PcapSocketProbeHandler) UnregisterProbe(n *graph.Node, e ProbeEventHand
...
 
 	probe.state.Store(common.StoppingState)
-	probe.listener.Close()
-
-	p.portAllocator.Release(probe.port)
+	err := probe.listener.Close()
+	if err != nil {
+		return err
+	}
+	err = p.portAllocator.Release(probe.port)
```
Comment (lebauce) [inline flow/probes/pcapsocket.go]: nit: return p.portAllocator.Release(probe.port)

**Comment by lebauce** [comment]
Ping @nplanel

**Comment by nplanel** [comment]
run tests

**Comment by nplanel** [comment]
run  tests

**Comment by nplanel** [comment]
run skydive-scale-tests

File: agent/agent.go [hunk 3]
```diff
@@ -83,7 +83,9 @@ func NewAnalyzerStructClientPool(authOpts *shttp.AuthenticationOpts) (*ws.Struct
 		if err != nil {
 			return nil, err
 		}
-		pool.AddClient(c)
+		if err := pool.AddClient(c); err != nil {
+			logging.GetLogger().Error("Pool add client ", err)
```
Comment (lebauce) [inline agent/agent.go]: logging.GetLogger().Errorf("Failed to add client to pool: %s", err)

File: flow/probes/pcapsocket.go [hunk 4]
```diff
@@ -132,11 +132,11 @@ func (p *PcapSocketProbeHandler) UnregisterProbe(n *graph.Node, e ProbeEventHand
 	p.Ctx.FTA.Release(probe.flowTable)
 
 	probe.state.Store(common.StoppingState)
-	probe.listener.Close()
-
-	p.portAllocator.Release(probe.port)
-
-	return nil
+	err := probe.listener.Close()
```
Comment (lebauce) [inline flow/probes/pcapsocket.go]: one liner !!!

if err := probe.listener.Close(); err != nil {
	return err
}

File: topology/probes/neutron/neutron.go [hunk 5]
```diff
@@ -257,8 +257,11 @@ func (p *Probe) updateNode(node *graph.Node, attrs *Metadata) {
 			tr.AddMetadata("PeerIntfMAC", attachedMac)
 		}
 	}
-	tr.Commit()
-
+	err := tr.Commit()
+	if err != nil {
+		p.Ctx.Logger.Error("Commit failed %+v : %v", tr, err)
```
Comment (lebauce) [inline topology/probes/neutron/neutron.go]: p.Ctx.Logger.Errorf("Commit failed: %s", err)

File: topology/probes/neutron/neutron.go [hunk 6]
```diff
@@ -300,7 +303,11 @@ func (p *Probe) updateNode(node *graph.Node, attrs *Metadata) {
 					if i == len(path)-1 {
 						tr.AddMetadata("PeerIntfMAC", attachedMac)
 					}
-					tr.Commit()
+					err := tr.Commit()
+					if err != nil {
+						p.Ctx.Logger.Error("Commit failed %+v : %v", tr, err)
```
Comment (lebauce) [inline topology/probes/neutron/neutron.go]: p.Ctx.Logger.Errorf("Commit failed: %s", err)

File: topology/probes/neutron/neutron.go [hunk 7]
```diff
@@ -363,7 +370,11 @@ func (p *Probe) OnNodeAdded(n *graph.Node) {
 			if uuid, _ := qvoNode.GetFieldString("ExtID.vm-uuid"); uuid != "" {
 				tr.AddMetadata("ExtID.vm-uuid", uuid)
 			}
-			tr.Commit()
+			err := tr.Commit()
+			if err != nil {
+				p.Ctx.Logger.Error("Commit failed %+v : %v", tr, err)
```
Comment (lebauce) [inline topology/probes/neutron/neutron.go]: p.Ctx.Logger.Errorf("Commit failed: %s", "err)

File: websocket/server.go [hunk 8]
```diff
@@ -166,7 +166,9 @@ func (s *Server) newIncomingClient(conn *websocket.Conn, r *auth.AuthenticatedRe
 	c.State.Store(common.RunningState)
 
 	// add the new Speaker to the server pool
-	s.AddClient(pc)
+	if err := s.AddClient(pc); err != nil {
+		return nil, fmt.Errorf("AddClient error: %s", err)
```
Comment (lebauce) [inline websocket/server.go]: return nil, fmt.Errorf("Failed to add client to server: %s", err)

**Comment by lebauce** [comment]
run skydive-compile-tests
run skydive-functional-tests-backend-elasticsearch
run skydive-functional-tests-backend-orientdb
run skydive-go-fmt
run skydive-scale-tests

**Comment by lebauce** [comment]
run skydive-scale-tests
run skydive-cdd-overview-tests

**Comment by lebauce** [comment]
run skydive-cdd-overview-tests
run skydive-scale-tests

**Comment by nplanel** [comment]
run skydive-scale-tests

**Comment by lebauce** [comment]
run skydive-cdd-overview-tests
run skydive-scale-tests

**Comment by lebauce** [comment]
run skydive-scale-tests

**Comment by lebauce** [comment]
run skydive-scale-tests

**Comment by lebauce** [comment]
run skydive-scale-tests

**Comment by lebauce** [comment]
run skydive-scale-tests

**Comment by lebauce** [comment]
run skydive-scale-tests

**Comment by nplanel** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by nplanel** [comment]
run skydive-functional-tests-backend-elasticsearch

**Pull request merged**